### PR TITLE
Feature/add create own identity attribute description

### PR DESCRIPTION
**Pull request by britsta**
Add scenario description for "Create own IdentityAttribute" -> Renaming to "Create Attribute for yourself" to be able to differentiate the scenario more clearly from the "Create a RepositoryAttribute" use case and also to be able to explain the creation of RelationshipAttributes

### CODE REVIEW

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:26]: `<...>` (use ` ); "insert" instead of "feed"

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:7]: Maybe a bit confusing, since "user" usually is associated with the App and not the Connector. Maybe write maintainer/developer/integrator or just omit it.

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:28]: I guess here and also later it would be nice link to a use case.

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:42]: Include as Lucidchart and delete svg.

Also I have some questions to the graphic itself: 
Again I find the differentiation between Connector and User confusing. I guess once more your are not referring to the App user, but to a person using the Connector. Then, maybe the names on the left side should be adjusted, e.g. lower part could be denoted with "enmeshed runtime" (in case that is the correct place of this process) or something else.
Furthermore, is there an error code for "You are not allowed to create an attribute that is not owned by yourself"? What will be the response if you'd try that?

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:5]: Since you are referring to an "http request" and not an "enmeshed Request" (like described in the Request and Response Introduction), I'd recommend to explicitly say so here to prevent confusion.

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:48]: An example of a simple IdentityAttribute is one of the type [`DisplayName`]({% link _docs_integrate/attribute-values.md %}#displayname). (I'd rather keep the sentence human-readable instead of technically precise.)

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:67]: An example of a complex IdentityAttribute is one of the type [`BirthDate`]({% link _docs_integrate/attribute-values.md %}#birthdate).

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:84]: as additional IdentityAttributes of type

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (britsta) [reply _includes/scenarios/integrate/create-own-identityattribute.md:42]: I will change "user" to "Integrator" as above.

As far as I know, there is currently no error code for the error "You are not allowed to create an attribute that is not owned by yourself". Also, if you'd try to do that, there is currently unexpected behaviour shown: You'd get "An unexpected error occurred" instead of "You are not allowed to create an attribute that is not owned by yourself" as wished. I've talked briefly with Julian about it already.

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (britsta) [reply _includes/scenarios/integrate/create-own-identityattribute.md:28]: I've replaced the "HTTP request using `GET /api/v2/Account/IdentityInfo`" part with a link to a use case now (it's secondary information here anyway). 
Also, I changed the position of the link to the Create an Attribute use case, so that it is in a more prominent place.

**Review by vinzlange — APPROVED** [review]

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (jkoenig134) [inline _includes/scenarios/integrate/create-own-identityattribute.md:3]: @britsta how to resolve these TODOs? It seems like these are the only thing blocking this PR.

Comment (vinzlange) [reply _includes/scenarios/integrate/create-own-identityattribute.md:3]: leave todo open, will be fixed in https://github.com/nmshd/documentation/pull/170

Comment (jkoenig134) [reply _includes/scenarios/integrate/create-own-identityattribute.md:3]: ok, great.

Comment (jkoenig134) [reply _includes/scenarios/integrate/create-own-identityattribute.md:3]: so then the only open thing is the duplicated information (tables)

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (jkoenig134) [inline _includes/scenarios/integrate/create-own-identityattribute.md:20]: It's a huge problem that we copy paste the address `id134nJmN7E4Carb6KyRJyePVnXxVHEYQgWD` everywhere. From time to time I recreate the connector b/c it gets to full with data and then I have to search the whole docs for this.

IMO we could  
1. just remove this and just tell how they find it out
2. insert an api snippet to query the address
3. load this with js

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (jkoenig134) [inline _includes/scenarios/integrate/create-own-identityattribute.md:18]: > Note that the properties `validFrom`, `validTo` and `tags` are optional, so you can omit them.

In the [other table](https://enmeshed.eu/integrate/data-model-overview#identityattribute) is all information (data type and optional or not). I think it's not good to write this in text form.

Comment (jkoenig134) [inline _includes/scenarios/integrate/create-own-identityattribute.md:18]: > as the [Create an Attribute]({% link _docs_use-cases/use-case-consumption-create-an-attribute.md %}) use case relates to the creation of Attributes for yourself and not for others
then I have a problem. Why do we need this page and also the other page if they actually do the same?

Comment (britsta) [reply _includes/scenarios/integrate/create-own-identityattribute.md:18]: I'm not sure if I'm the right person to answer this question in principle, as I didn't originally set up the various scenarios and use cases. But as I understand it, the purpose of the scenarios is to explain how to apply certain use cases. In this scenario, the "Create an Attribute" use case plays a central role. However, you also have the opportunity on this scenario page to mention how you can obtain the required Address of the Connector (use case "Get currently used Identity"), which Attribute Values you can use and where you can find them, or the differences in Attribute creation between simple and complex Attributes.

So I understand this scenario page as a way to connect information that supports someone in applying the use case and that goes beyond the technical documentation of the use case.

Comment (jkoenig134) [reply _includes/scenarios/integrate/create-own-identityattribute.md:18]: That's good! But I think we have a naming clash if we have to explain this here.

Comment (britsta) [reply _includes/scenarios/integrate/create-own-identityattribute.md:18]: I have now seen in the runtime code that the use cases have been restructured/renamed and the "CreateAttribute" use case here should be replaced by the "CreateIdentityAttribute" use case. From this perspective, do you think it is a naming clash that the use case and the scenario have the same name, and if so, how would you recommend changing this issue?

Comment (jkoenig134) [reply _includes/scenarios/integrate/create-own-identityattribute.md:18]: Yep, it was changed. But it still only allows to create own IdentityAttributes.

Comment (britsta) [reply _includes/scenarios/integrate/create-own-identityattribute.md]: Now that I have replaced the table with a JSON payload, I have the following general question: 

If I give an example JSON payload and want to emphasise that some of the properties are optional and can be omitted, would you find it okay to point this out in text form then? With the tables, I had the option of adding an extra "Required" column, with the JSON payloads it is more difficult to indicate directly whether a property is optional.

Comment (britsta) [reply _includes/scenarios/integrate/create-own-identityattribute.md]: I've changed the sentence now so that it's clear that only own IdentityAttributes can be created.

As the documentation for the "Create an attribute" use case will change with #179 and this scenario page now already refers to the new documentation/functionality, it may be necessary to wait until #179 is published before merging this scenario

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:7]: `content`

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:27]: Not needed anymore.

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:32]: Mention the term "child Attribute"/"ChildAttribute" here.

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:34]: I suppose this diagram needs to be adjusted to not have the separate chack for the owner. Also, for the first cross section I'd reformulate the text to "**Valid** input entered for creating an IdentityAttribute?". And writing about complex Attributes, I'd again use the term child Attribute.

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:44]: omit owner

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:38]: `content`

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:53]: doesn't have more than 100 characters

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:57]: `content`

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:63]: omit owner

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:66]: In another PR you wrote "" around the values of the child attributes. I commented on it and you argued that they can be specified like this. I'm not sure anymore, what was the outcome of that conversation, but perhaps check, that we have it consistently in the scenarios.

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:74]: Maybe you can use the term child Attribute here again.

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-own-identityattribute.md:78]: Rather link to the now merged Share own Attribute to peer scenario. Perhaps you can mention it in the for of a "What's next?" section, as you did for the other scenarios.

File: _includes/scenarios/integrate/create-own-identityattribute.md
Comment (Milena-Czierlinski) [reply _includes/scenarios/integrate/create-own-identityattribute.md:66]: Ok, I just saw that this is consistent with the merged scenarios.

File: _includes/scenarios/integrate/create-attribute-for-yourself.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-attribute-for-yourself.md:95]: It's an interesting fact, that here only the peer is allowed as owner and that it's the only case that has this restriction. Maybe state explicitly in the other cases, that both you and the peer can be the owner.

File: _includes/scenarios/integrate/create-attribute-for-yourself.md
Comment (Milena-Czierlinski) [inline _includes/scenarios/integrate/create-attribute-for-yourself.md:89]: I'd prefer this to be above "Utilization of a ReadAttributeRequestItem", since I guess it makes sense for a reader, who wants to create a specific Attribute, to read this first.

**Review by Milena-Czierlinski — COMMENTED** [review]
Just some minor things mostly regarding language. Otherwise, this looks really good.

File: _includes/scenarios/integrate/create-attribute-for-yourself.md
Comment (britsta) [reply _includes/scenarios/integrate/create-attribute-for-yourself.md:21]: I understand your point of view. At this point, I'd prefer to use the order of the properties as it is in the Data Model Overview for the IdentityAttribute to be consistent.

Comment (Milena-Czierlinski) [reply _includes/scenarios/integrate/create-attribute-for-yourself.md:21]: Then how about adjusting it in the Data Model Overview, too, such that the required properties are on top?

Comment (britsta) [reply _includes/scenarios/integrate/create-attribute-for-yourself.md:21]: I think the order of the properties in the Data Model Overview comes from how the IdentityAttribute is defined with the help of the AbstractAttribute in the content library. From this perspective, the current order in the Data Model Overview makes sense to me. However, I also think that it would be a legitimate order of the properties to sort by relevance (for example, required or optional). IMO we should have to take a look at this for the other data object types as well and then make a standardised decision.

File: _includes/scenarios/integrate/create-attribute-for-yourself.md
Comment (britsta) [reply _includes/scenarios/integrate/create-attribute-for-yourself.md:34]: "components" replaced by "properties". 👍

Regarding the term "child Attribute", I would prefer not to use it in the documentation, and simply use "child" instead, so that it has less neologism vibes.

Comment (Milena-Czierlinski) [reply _includes/scenarios/integrate/create-attribute-for-yourself.md:34]: I must admit that I don't see your "neologism vibes" point, since having an Attribute that is referred to as "child" isn't too far from calling it a "child Attribute", but I don't mind if you just call them "children".

File: _includes/scenarios/integrate/create-attribute-for-yourself.md
Comment (britsta) [reply _includes/scenarios/integrate/create-attribute-for-yourself.md:82]: I like it when the last regular sentence is "...a CreateAttributeRequestItem, ReadAttributeRequestItem, ProposeAttributeRequestItem or ShareAttributeRequestItem should be used for this", so that it fits well with the subsection headers. I also feel it is a rather technical side aspect that is not so central here compared to the same topic in the IdentityAttributes section, but it is good to know to make a better comparison with the IdentityAttributes. No strong opinion on this, I think there are several good options and each has its advantages

File: _includes/scenarios/integrate/create-attribute-for-yourself.md
Comment (britsta) [reply _includes/scenarios/integrate/create-attribute-for-yourself.md:95]: I have also added the information about the owner to the CreateAttributeRequestItem 👍  

With the ReadAttributeRequestItem and ShareAttributeRequestItem this is not easily done in one sentence, as in these cases a third party is also possible as the owner and I'd not like to hide this fact when I start talking about the owner. Preference for leaving out the owner part or going into more detail (or no strong opinion)?

File: _includes/scenarios/integrate/create-attribute-for-yourself.md
Comment (Milena-Czierlinski) [reply _includes/scenarios/integrate/create-attribute-for-yourself.md:82]: Ok, I agree that there are several options. Decide on what suits you best.

File: _includes/scenarios/integrate/create-attribute-for-yourself.md
Comment (Milena-Czierlinski) [reply _includes/scenarios/integrate/create-attribute-for-yourself.md:95]: If it doesn't get too cluttered, I'd state the allowance of a ThirdParty owner here then, too.

**Review by Milena-Czierlinski — APPROVED** [review]

**Pull request merged**
